src/pages/registration.tsx

```diff
@@ -43,35 +50,105 @@ const InteractionsContainer = styled.div`
   width: 100%;
 `;
 
+const DropIcon = styled.img`
+  width: 4rem;
+`;
+
+interface UserForm {
+  firstName: string;
+  lastName: string;
+  gender: string;
+}
+
+interface FormEntry {
+  field: string;
+  value: string;
+}
+
+const emptyForm: UserForm = {
+  firstName: '',
+  lastName: '',
+  gender: ''
+};
+
+const formReducer = (state: UserForm, { field, value }: FormEntry) => ({
```

Comment:
```suggestion
const formReducer = (state: UserForm, entry: [field: string, value: string]) => ({
```
You can use named (or unnamed) tuples instead of an interface. It'll make calling it a lot simpler. No need to construct an object, just pass in the tuple.

Reply:
Updated for now